### ISSUE TITLE
能否将 dashboard 的 SentinelApiClient 抽象出来？

### ISSUE THREAD

**Issue by waveng** [alibaba/Sentinel#191]
## Issue Description

能否将 dashboard 的 SentinelApiClient 抽象出来，这样便可以实现 dashboard 端动态规则的读取和推送 自定义，让使用者根据自己 DataSource 方案，去实现 SentinelApiClient，这样我们 dashboard 这边升级后使用者可以无缝升级到最新的 dashboard 

### Describe what happened (or what feature you want)
目前的 dashboard 的 SentinelApiClient 是直接通过 http 从客户端读取、推送数据，无法灵活的自定义实现读取、推送，只能直接的 dashboard 上修改，这样会导致无法无缝升级到最新的 dashboard

### Describe what you expected to happen
我希望可以实现自定义dashboard端的读取、推送数据，并且可以无缝升级到最新的 dashboard

### Anything else we need to know?
我这里已经实现了, 请见：[custom-dashboard](https://github.com/waveng/Sentinel/tree/custom-dashboard)。

这里我将 SentinelApiClient 抽象为 [sentinel-dashboard-datasource](https://github.com/waveng/Sentinel/tree/custom-dashboard/sentinel-dashboard-support/sentinel-dashboard-datasource)

并且实现了zookeeper的推送方案 [sentinel-dashboard-datasource-zookeeper](https://github.com/waveng/Sentinel/tree/custom-dashboard/sentinel-dashboard-support/sentinel-dashboard-datasource-zookeeper)，项目有单元测试

要想使用 sentinel-dashboard-datasource-zookeeper 需要[sentinel-datasource-zookeeper](https://github.com/waveng/Sentinel/tree/custom-dashboard/sentinel-extension/sentinel-datasource-zookeeper)包的支持，需要再客户引用该包，并参照 sentinel-demo-zookeeper-datasource 进行配置；完整的使用实例见[test-dashboard-zk](https://github.com/waveng/Sentinel/tree/test-dashboard-zk)中的 [sentine-demo-dubbo-zookeeper](https://github.com/waveng/Sentinel/tree/test-dashboard-zk/sentinel-demo/sentine-demo-dubbo-zookeeper)

**Comment by sczyh30** [comment]
Good idea. We may do some abstraction on some universal functions later.

**Comment by waveng** [comment]
Thank you to adopt

**Comment by liangyuanpeng** [comment]
any update?
